src/components/Search/SearchHeader.tsx

```diff
@@ -0,0 +1,133 @@
+import * as Illustrations from '@components/Icon/Illustrations';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as SearchActions from '@libs/actions/Search';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import type {SearchQuery, SelectedTransactions} from '@src/types/onyx/SearchResults';
+import type DeepValueOf from '@src/types/utils/DeepValueOf';
+import type IconAsset from '@src/types/utils/IconAsset';
+
+type SearchHeaderProps = {
+    query: SearchQuery;
+    selectedItems?: SelectedTransactions;
+    clearSelectedItems?: () => void;
+    hash: number;
+};
+
+type SearchHeaderOptionValue = DeepValueOf<typeof CONST.SEARCH_BULK_ACTION_TYPES> | undefined;
+
+function SearchHeader({query, selectedItems = {}, hash, clearSelectedItems}: SearchHeaderProps) {
+    const {translate} = useLocalize();
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const headerContent: {[key in SearchQuery]: {icon: IconAsset; title: string}} = {
+        all: {icon: Illustrations.MoneyReceipts, title: translate('common.expenses')},
+        shared: {icon: Illustrations.SendMoney, title: translate('common.shared')},
+        drafts: {icon: Illustrations.Pencil, title: translate('common.drafts')},
+        finished: {icon: Illustrations.CheckmarkCircle, title: translate('common.finished')},
+    };
+
+    const getHeaderButtons = () => {
+        const options: Array<DropdownOption<SearchHeaderOptionValue>> = [];
+        const selectedItemsKeys = Object.keys(selectedItems ?? []);
+
+        if (selectedItemsKeys.length === 0) {
+            return null;
+        }
+
+        const itemsToDelete = selectedItemsKeys.filter((id) => selectedItems[id].canDelete);
+
+        if (itemsToDelete.length > 0) {
+            options.push({
+                icon: Expensicons.Trashcan,
+                text: translate('search.bulkActions.delete'),
+                value: CONST.SEARCH_BULK_ACTION_TYPES.DELETE,
+                onSelected: () => {
+                    clearSelectedItems?.();
+                    SearchActions.deleteMoneyRequestOnSearch(hash, itemsToDelete);
+                },
+            });
+        }
+
+        // @TODO: Uncomment when actions are ready
+
+        // const itemsToHold = selectedItemsKeys.filter((id) => selectedItems[id].action === CONST.SEARCH_BULK_ACTION_TYPES.HOLD);
+
+        // if (itemsToHold.length > 0) {
+        //     options.push({
+        //         icon: Expensicons.Stopwatch,
+        //         text: translate('search.bulkActions.hold'),
+        //         value: CONST.SEARCH_BULK_ACTION_TYPES.HOLD,
+        //         onSelected: () => {
+        //             SearchActions.holdMoneyRequestOnSearch(hash, itemsToHold, '');
```

Comment:
We need to pass a hold reason as 3rd param here. I think @Kicu is working on the hold reason page as part of https://github.com/Expensify/App/issues/39890